File: 1878-check-if-array-is-sorted-and-rotated/1878-check-if-array-is-sorted-and-rotated.cpp
class Solution {
public:
    bool check(vector<int>& nums) {
        int n = nums.size();
        int rotation = 0;
        for(int i = 0; i<n; i++){
            if(nums[i] > nums[(i+1)%n]){
                rotation++;
            }
        }

        return rotation <= 1;
    }
};

File: 1878-check-if-array-is-sorted-and-rotated/1878-check-if-array-is-sorted-and-rotated.java
class Solution {
    public boolean helpr(int[] nums, int s, int e){
        for(int i = s; i<e; i++){
            if(nums[i] > nums[i+1])
                return false;
        }
        return true;
    }
    public boolean check(int[] nums) {
        int ind = 0;
           while(ind < nums.length && nums[0] >= nums[nums.length-1]){
               int t = nums[0];
               for(int i = 0; i<nums.length-1; i++){
                   nums[i] = nums[i+1];
               }
               nums[nums.length-1] = t;
               ind++;
           }
            return helpr(nums, 0, nums.length-1);
       }
}